src/test/kotlin/com/igorwojda/integer/digitfrequency/challenge.kt

```diff
@@ -4,7 +4,22 @@ import org.amshove.kluent.shouldBeEqualTo
 import org.junit.jupiter.api.Test
 
 private fun equalDigitFrequency(i1: Int, i2: Int): Boolean {
-    TODO("not implemented")
+    val digitMap = mutableMapOf<Int, Int>()
```

Comment:
Interesting solution. How would have solved this if it required to compare two strings and check if they had an equal character frequency?

Reply:
If the function were to check the character frequency I would use the following extensions:
`private fun characterFrequency(i1: String, i2: String): Boolean {`
`return i1.toList().groupBy { it } == i2.toList().groupBy { it }`
`}`
By writing this and exploring the documentation I learned that the Map.equals() compares by default the size, entries and keys of the two maps

Reply:
👍 
And what condition could you use to have an early return?